packages/openid4vp/src/openid4vp-auth-request/validate-openid4vp-auth-request.ts

```diff
@@ -0,0 +1,57 @@
+import { Oauth2Error } from '@openid4vc/oauth2'
+import type { Openid4vpAuthRequest } from './v-openid4vp-auth-request'
+
+/**
+ * Validate the OpenId4Vp Authorization Request parameters
+ */
+export const validateOpenid4vpAuthRequestParams = (
+  params: Openid4vpAuthRequest,
+  options: {
+    wallet?: {
+      nonce?: string
+    }
+  }
+) => {
+  if (!params.redirect_uri && !params.response_uri) {
+    throw new Oauth2Error('OpenId4Vp Authorization Request redirect_uri or response_uri is required.')
+  }
+
+  if (params.response_uri && !['direct_post', 'direct_post.jwt'].find((mode) => mode === params.response_mode)) {
+    throw new Oauth2Error(
+      `OpenId4Vp Authorization Request response_mode must be direct_post or direct_post.jwt when response_uri is provided. Current: ${params.response_mode}`
+    )
+  }
+
+  if (
+    [params.presentation_definition_uri, params.presentation_definition, params.dcql, params.scope].filter(Boolean)
+      .length > 1
+  ) {
+    throw new Oauth2Error(
+      'Exactly one of the following parameters MUST be present in the Authorization Request: dcql_query, presentation_definition, presentation_definition_uri, or a scope value representing a Presentation Definition.'
+    )
+  }
+
+  if (params.request_uri_method && !params.request_uri) {
+    throw new Oauth2Error(
+      'OpenId4Vp Authorization Request request_uri_method parameter MUST NOT be present if the request_uri parameter is not present.'
+    )
+  }
+
+  if (params.trust_chain && !params.client_id.startsWith('http://') && !params.client_id.startsWith('https://')) {
```

Comment:
I guess http:// is only allowed when allowUnsecureHttpUrl is enabled right? I have handled with the `vHttpsUrl` validation.